mithril-aggregator/src/services/certificate_chain_synchroniser/synchroniser_service.rs

```diff
@@ -0,0 +1,735 @@
+        }
+    }
+
+    async fn retrieve_and_validate_remote_certificate_chain(
+        &self,
+        starting_point: Certificate,
+    ) -> StdResult<Vec<Certificate>> {
+        // IMPORTANT: Order matters, certificates must be ordered from genesis to latest
+        // (fetched database data is returned from last inserted to oldest)
+        let mut validated_certificates = VecDeque::new();
+        let mut certificate = starting_point;
+
+        loop {
+            let parent_certificate = self
+                .certificate_verifier
+                .verify_certificate(&certificate, &self.genesis_verifier.to_verification_key())
+                .await
+                .with_context(
+                    || format!("Failed to verify certificate: `{}`", certificate.hash,),
+                )?;
+
+            match parent_certificate {
+                None => {
+                    validated_certificates.push_front(certificate);
+                    break;
+                }
+                Some(parent) => {
+                    // At the start of the retrieval the first certificate may not be the first of
+                    // its epoch, filter them out since we only need one certificate per epoch
+                    if !validated_certificates.is_empty() || parent.epoch != certificate.epoch {
+                        validated_certificates.push_front(certificate);
+                    }
+
+                    certificate = parent;
+                }
+            }
+        }
+
+        Ok(validated_certificates.into())
+    }
+
+    async fn store_certificate_chain(&self, certificate_chain: Vec<Certificate>) -> StdResult<()> {
+        self.certificate_storer
+            .insert_or_replace_many(certificate_chain)
+            .await?;
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl CertificateChainSynchronizer for MithrilCertificateChainSynchronizer {
+    async fn synchronize_certificate_chain(&self, force: bool) -> StdResult<()> {
+        debug!(self.logger, ">> synchronize_certificate_chain"; "force" => force);
+
+        let sync_state = self.check_sync_state(force).await.with_context(|| {
+            format!("Failed to check if certificate chain should be sync (force: `{force}`)")
+        })?;
+        if sync_state.should_sync() {
+            info!(self.logger, "Start synchronizing certificate chain"; "sync_state" => ?sync_state);
+        } else {
+            info!(self.logger, "No need to synchronize certificate chain"; "sync_state" => ?sync_state);
+            return Ok(());
+        }
+
+        let starting_point = self
+            .remote_certificate_retriever
+            .get_latest_certificate_details()
+            .await?
+            .ok_or(
+                anyhow!("Remote aggregator doesn't have a chain yet")
+                    .context("Failed to retrieve latest remote certificate details"),
+            )?;
+        let open_message = prepare_open_message_to_store(&starting_point);
```

Comment:
nitpick: should be computed from the retrieved chain lastest certificate (as the starting point may be eluded from the chain to store if it's not the first of its epoch)